src/AI/WallyAI.java

```diff
@@ -0,0 +1,788 @@
+package AI;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderAbility;
+import Engine.GameAction;
+import Engine.GameAction.ActionType;
+import Engine.GameActionSet;
+import Engine.Path;
+import Engine.Utils;
+import Engine.XYCoord;
+import Engine.Combat.CombatEngine;
+import Terrain.GameMap;
+import Terrain.Location;
+import Terrain.TerrainType;
+import Units.Unit;
+import Units.UnitModel;
+import Units.UnitModel.UnitEnum;
+import Units.Weapons.Weapon;
+import Units.Weapons.WeaponModel;
+
+/**
+ *  Wally makes(?) and breaks walls.
```

Comment:
(?)?  Are you not sure what Wally does? A little more info in general might be useful.
At some point, I might want to allow for short AI descriptions from the CO select screen just to provide the gist of what to expect.

Reply:
I tried to make him make walls, but I'm not so sure he's good at it, haha.
I'm not sure what exactly *should* go here, and to what degree of precision.

I agree it would be cool, yeah. Could be a COInfo page, possibly? A tooltip might be better, though.

Reply:
> I'm not sure what exactly should go here
Whatever was your guiding principle in constructing this AI. ISAI focuses on infantry, Spender tries to spend everything, Muriel builds units to counter whatever the opponent has. Wally's current phrase "makes(?) and breaks walls" is not very clear. As in, he tries to form proper British-style battle lines and prevent his opponent from doing likewise? No gory algorithm details of course; just a high-level description of the way decisions are made, or the value function(s) in play.

Yeah, I might lean towards either a floating box at the bottom of the COSelect screen with a blurb about whatever is highlighted, or a Q-menu a la the CO Info page. This is the wrong PR for that discussion though.

Reply:
He attempts to meatshield his strongest assets, and tries to only commit units when he can get as much value as he has committed.

It doesn't entirely work, but I'm not really interested in poking at it more until gameSetup is merged 'cause I'm tired of hacking in AI.